pkg/karmadactl/cmdinit/kubernetes/deploy.go

```diff
@@ -613,6 +614,20 @@ func (i *CommandInitOption) karmadaAggregatedAPIServerImage() string {
 	return i.KarmadaAggregatedAPIServerImage
 }
 
+// get image pull secret
+func (i *CommandInitOption) getImagePullSecrets() []corev1.LocalObjectReference {
+	var imagePullSecrets []corev1.LocalObjectReference
+	if i.PullSecrets != nil {
```

Comment:
This check is actually redundant because the `for .. range` can work with nil slice.